### PR TITLE
Fix resource access requests for apps

### PR DESCRIPTION
**Pull request by nklaassen**
Resource access requests for apps are currently broken for static apps because I incorrectly assumed they would be returned from calls to `GetApp`, which is not the case.

The change here is to use `ListResources` instead of `GetApp`, `GetNode`, etc... in `PruneResourceRequestRoles`. This is more consistent with the way we search for resources and build the access request in the first place with `tsh request search` and from the web UI, where `ListResources` is already used.

### CODE REVIEW

File: lib/services/access_request.go [hunk 1]
```diff
@@ -1450,55 +1449,82 @@ func roleAllowsResource(
...
+	}
+	return strings.Join(matchers, " || ")
+}
+
+// FixListResourcesType returns the value to use for ResourceType in a
+// ListResourcesRequest based on the kind of resource you're searching for.
+// Necessary because some resource kinds don't support ListResources directly,
+// so you have to list the parent kind. Use FixListResourcesResult to map back
+// to the given kind.
+func FixListResourcesType(kind string) string {
```
[the remark this answers is not in the document]
Comment (nklaassen) [reply lib/services/access_request.go:1493]: I'll rename. I pulled this out of tool/tsh/access_request.go, it's called there

[the remark this answers is not in the document]
Comment (nklaassen) [reply lib/services/access_request.go]: renamed these in [625d7e4](https://github.com/gravitational/teleport/pull/13955/commits/625d7e42c9b35a766a4a52b26cd0977679c35745)

[the remark this answers is not in the document]
Comment (ravicious) [reply lib/services/access_request.go]: Good change! 👍

**Review by ravicious — APPROVED** [review]

**Review by jakule — APPROVED** [review]

File: lib/services/access_request.go [hunk 2]
```diff
@@ -1450,55 +1449,85 @@ func roleAllowsResource(
...
-					resources = append(resources, kubeV3)
+			resources = append(resources, leafResources...)
+		}
+	}
+	return resources, nil
+}
+
+// namesMatcher returns a PredicateExpression which matches any of a given list
+// of names. Given names will be escaped and quoted when building the expression.
+func namesMatcher(names []string) string {
```
[the remark this answers is not in the document]
Comment (nklaassen) [reply lib/services/access_request.go]: I went with `anyNameMatcher` so it doesn't sound like it might return some names OR a matcher 🤷‍♂️

**Pull request merged**